stub/src/main/java/io/grpc/stub/CallStreamObserver.java

```diff
@@ -99,4 +99,19 @@
    * @param count more messages
    */
   public abstract void request(int count);
+
+  /**
+   * Sets message compression for subsequent calls to {@link #onNext}.
+   *
+   * @param enable whether to enable compression.
+   */
+  public abstract void setMessageCompression(boolean enable);
+
+  /**
+   * Sets the compression algorithm to use for the call.  Must be called before sending any
+   * messages.
+   *
+   * @param compression the compression algorithm to use.
+   */
+  public abstract void setCompression(String compression);
```

Comment:
This looks like it should be in ServerCallStreamObserver. It can't be implemented on client-side, because client-side uses `CallOptions` instead.